### ISSUE TITLE
Speed up sersic + check gradients

### ISSUE THREAD

**Issue by mjyb16** [mjyb16/supermage#5]
In the sersic mge models (with and without gas), the following code occurs in the forward pass:

```
        surf=torch.zeros(self.N_components, device=device, dtype=dtype)

        for i in range(self.N_components):
            surf[i]=interp1d(self.n_grid,self.surf_grid[:,i],n,extend="extrapolate")
```

I believe this slows the code down a lot and may even break gradients due to re-initializing the sersic. The optimal solution is to move this code block into the init, save the interp1d as a class attribute, and then access the interpolator from the forward pass.
